### PR TITLE
Fixes #9910: Bugs on datasource initilization and update

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/9910

### CODE REVIEW

File: rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala [hunk 1]
```diff
@@ -397,6 +397,11 @@ object RudderConfig extends Loggable {
   val historizationRepository : HistorizationRepository =  historizationJdbcRepository
   val roApiAccountRepository : RoApiAccountRepository = roLDAPApiAccountRepository
   val woApiAccountRepository : WoApiAccountRepository = woLDAPApiAccountRepository
+  lazy val dataSourceRepository = new DataSourceRepoImpl(
```
Comment (ncharles) [inline rudder-web/src/main/scala/bootstrap/liftweb/AppConfig.scala]: this lazy val is in the middle of non lazy val - could you move it with others lazy val ?

File: rudder-core/src/main/scala/com/normation/rudder/datasources/DataSourceRepository.scala [hunk 2]
```diff
@@ -133,6 +133,16 @@ class DataSourceRepoImpl(
 
   private[this] var datasources = Map[DataSourceId, DataSourceScheduler]()
 
+  // Initialize data sources scheduler, with all sources present in backend
+  getAll match {
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/datasources/DataSourceRepository.scala]: I think it should not be done at class instanciation (but we have to document that the start is not automatic). Not having it hiden in the class instanciation allows to split:

- error from instanciation (most likelly code problem, and hard to debug: "no such class error" most of the time) from error due to boot (most likelly connection pb)
- let the user choose when to trigger datasource fetch (in a future version of Rudder, we may want to boot completly and as fast as possible before starting datasources schedules)

Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/datasources/DataSourceRepository.scala]: And so, in Rudder the solution is to use a ```BootstrapChecks``` (that will give nice init log for free)

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**